app/src/main/java/org/simple/clinic/editpatient/PatientEditScreenController.kt

```diff
@@ -168,4 +173,90 @@ class PatientEditScreenController @Inject constructor(
         .filter { it.isNotEmpty() }
         .map { errors -> { ui: Ui -> ui.hideValidationErrors(errors) } }
   }
+
+  private fun savePatientDetails(events: Observable<UiEvent>): Observable<UiChange> {
```

Comment:
I'd like another pair of eyes on this function, please. It seems to be doing a lot, but I'm not sure how I can simplify this further.